cfn-resources/online-archive/cmd/resource/resource.go

```diff
@@ -43,31 +42,22 @@ var requiredCriteriaType = map[string][]string{
 	"CUSTOM": {"Query"},
 }
 
-func setup() {
+func setupLogAndProfile(currentModel *Model) {
```

Comment:
nit: Could we have a `setupLogs` and `setupProfile`? Method names with `And` inside smell like an aggregation of responsibility that will likely increase in the future.

Reply:
I see your point but they're really very small methods, almost one line each.  What about keeping the name "setup", at the end the responsibility is to setup the methods, I wanted to be more explicit about what is set up but maybe it's not needed.

Reply:
"setup()" methods are so generic that will basically always force you to go look inside what is doing.

Understand your concern of "it's just one line", but I guess the reason why we are aggregating this into separate funcs is to remove boilerplate code repeated across the four methods and make it easy to read, is that correct? 

I'll leave that up to you, if you think that the option of having separate dedicated methods won't be useful then I don't have strong opinions.

Reply:
I like the idea of having a separate```setDefaultProfileIfNotDefined```, and this function can be defined in a common package as all resources will make use of the same implementation.

Reply:
I like that